azure/converters/managedagentpool_test.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package converters
+
+import (
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2021-05-01/containerservice"
+	"github.com/Azure/go-autorest/autorest/to"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/cluster-api-provider-azure/azure"
+)
+
+func Test_AgentPoolToManagedClusterAgentPoolProfile(t *testing.T) {
+	cases := []struct {
+		name   string
+		pool   azure.AgentPoolSpec
+		expect func(*GomegaWithT, containerservice.ManagedClusterAgentPoolProfile)
+	}{
+		{
+			name: "Should set all values correctly",
+			pool: azure.AgentPoolSpec{
+				SKU:               "Standard_D2s_v3",
+				OSDiskSizeGB:      100,
+				Replicas:          2,
+				Version:           to.StringPtr("1.22.6"),
+				VnetSubnetID:      "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups//providers/Microsoft.Network/virtualNetworks//subnets/",
```

Comment:
Nit: should there be double slashes for `/resourceGroups//providers` and `virtualNetworks//subnets/"`?

Reply:
there should not be, I will fix it. Thanks